pytests/test_wff_parental_tax_credit.py

```diff
@@ -22,13 +22,161 @@ class TestKey(Reasoner):
+
+
+class TestWFF_ParentalLessChildcare(TestKey):
+    body = {
+        "applicant": {
+            "isPrincipalCarerForProportion": 4,
+            "receivesIncomeTestedBenefit": False
+        },
+        "threshold": {
+            "income": {
+                "workingForFamiliesParentalTaxCredit": True
+            }
+        }
+    }
+
+    def test_reasoning(self):
+        self.assertTrue(self.is_forbidden)
+
+
+class TestWFF_ParentalTaxCreditOnMTBenefit(TestKey):
+    body = {
+        "applicant": {
+            "isPrincipalCarerForProportion": 100,
+            "receivesIncomeTestedBenefit": True
+        },
+        "threshold": {
+            "income": {
+                "workingForFamiliesParentalTaxCredit": True
+            }
+        }
+    }
+
+    def test_reasoning(self):
+        self.assertTrue(self.is_forbidden)
+
+
+class TestWFF_ParentalTaxCreditForbiddenOnAcc(TestKey):
+    body = {
+        "applicant": {
+            "isPrincipalCarerForProportion": 34,
+            "receivesIncomeTestedBenefit": False,
+            "isOnACCCompensation": True
+        },
+        "threshold": {
+            "income": {
+                "workingForFamiliesParentalTaxCredit": True
+            }
+        }
+    }
+
+    def test_reasoning(self):
+        self.assertTrue(self.is_forbidden)
+
+
+class TestWFF_ParentalTaxCreditForbiddenOrphans(TestKey):
+    body = {
+        "applicant": {
+            "isPrincipalCarerForProportion": 34,
+            "receivesIncomeTestedBenefit": False,
+            # add the orphans bene requirements
+            "Age": 87,
+            "isNZResident": True,
+            "isParent": False,
+            "isPrincipalCarerForOneYearFromApplicationDate": True,
+            "normallyLivesInNZ": True
+        },
+        "child": {
+            "isDependent": True
+        },
+        "parents": {
+            "areDeceasedMissingOrIncapable": True
+        },
+        "threshold": {
+            "income": {
+                "workingForFamiliesParentalTaxCredit": True
+            }
+        }
+    }
+
+    def test_reasoning(self):
+        # Orphans benefit is permitted, so WFF PTC is not
+        self.assertTrue(self.isPermitted('isOrphansBenefit'))
+        self.assertTrue(self.is_forbidden)
+
+
+class TestWFF_ParentalTaxCreditForbiddenStudents(TestKey):
+    body = {
+        "applicant": {
+            "isPrincipalCarerForProportion": 34,
+            "receivesIncomeTestedBenefit": False,
+            # add the student allowance requirements
+            "isNZResident": True,
+            "normallyLivesInNZ": True,
+            "Age": 25,
+            "isStudyingFullTime": True,
+        },
+        "threshold": {
+            "income": {
+                "workingForFamiliesParentalTaxCredit": True
+            }
+        }
+    }
+
+    def test_reasoning(self):
+        # Orphans benefit is permitted, so WFF PTC is not
+        self.assertTrue(self.isPermitted('isStudentAllowance'))
```

Comment:
isStudentAllowance == orphans benefit??? naming makes your comment unclear

Reply:
Good catch, My comment is wrong